Scripts/Runtime/Data/Collections/UnsafeTypedStream.cs

```diff
@@ -747,6 +853,88 @@ internal void ReadBlock(byte* dstStart, int dstStartIndex, int elementsToRead)
                 return ref UnsafeUtility.AsRef<T>(m_ReaderHead);
             }
         }
+        
+        //*************************************************************************************************************
+        // ENUMERATOR
+        //*************************************************************************************************************
+        
+        [BurstCompile]
+        public struct Enumerator : IEnumerator<T>
+        {
+            [NativeDisableUnsafePtrRestriction] private readonly BufferInfo* m_BufferInfo;
+            private int m_LaneIndex;
+            private LaneReader m_LaneReader;
+            private int m_ArrayIndex;
+            private T m_Current;
+            
+            public T Current
+            {
+                get => m_Current;
+            }
+            
+            internal Enumerator(BufferInfo* bufferInfo)
+            {
+                m_BufferInfo = bufferInfo;
+                m_LaneIndex = 0;
+                m_ArrayIndex = -1;
+                m_LaneReader = new LaneReader(m_BufferInfo, m_LaneIndex);
+                m_Current = default;
+            }
+
+            public bool MoveNext()
+            {
+                //Progress the array index
+                m_ArrayIndex++;
+                //So long as our lane reader has an entry we're good
+                if (m_ArrayIndex < m_LaneReader.Count)
+                {
+                    m_Current = m_LaneReader.Read();
+                    return true;
+                }
+                
+                //Otherwise we need to find the next lane that has something
+                m_LaneIndex++;
+                for (; m_LaneIndex < m_BufferInfo->LaneCount; ++m_LaneIndex)
+                {
+                    //Update lane reader and array index
+                    m_LaneReader = new LaneReader(m_BufferInfo, m_LaneIndex);
+                    m_ArrayIndex = 0;
+                    //If the lane doesn't have anything in it, skip
+                    if (m_ArrayIndex >= m_LaneReader.Count)
+                    {
+                        continue;
+                    }
+
+                    m_Current = m_LaneReader.Read();
+                    return true;
+                }
+                
+                //We've gone through all lanes and nothing is left, we're done
+                m_Current = default;
+                return false;
+            }
+
+            public void Reset()
+            {
+                m_LaneIndex = 0;
+                m_ArrayIndex = -1;
+                m_LaneReader = new LaneReader(m_BufferInfo, m_LaneIndex);
+                m_Current = default;
+            }
+            
+            //We don't want anyone calling this in Burst since it will box the current value.
+            //But maybe it's useful for debugging purposes.
+            [BurstDiscard]
```

Comment:
I'm not 100% sure what actually happens here.

Originally i threw an exception in this function because I don't want you to accidentally write some code that tries to use this and you just don't know until your in a late night debugging/profiling session. 

But in that "in-progress" PR, https://github.com/decline-cookies/anvil-unity-dots/pull/55#discussion_r961083591, you had mentioned it might be better to leave in but discard the method in Burst so we don't normally get to use it and therefore would get an error at runtime that no such method exists... or maybe the Burst compiler would yell at us? 

But with Burst off, we might want to use it for some debugging. 

My guess is that this function call only actually gets resolved at runtime.